docs/enhancements/fep-000-api-device-fleet.md

```diff
@@ -0,0 +1,592 @@
+with the updated source reference.
+
+If an error happens during the creation of the `TemplateVersion` object, the fleet controller will
+mark a status condition on the fleet object (e.g. missing sources, missing floating tags, etc..),
+and the fleet controller will not proceed with the rollout.
+
+Once a `TemplateVersion` is created the fleet controller starts rolling out that update to devices.
+This update can take a long time (depending on the upgrade strategy). The fleet controller marks
+the latest template version that is being rolled out, as a `fleet-controller/templateVersion`
+annotation in the fleet for observability through the API.
+
+
+When a template is applied to a specific device via the Device API:
+* Floating tags for external references (os image, git refs, etc..) get converted to stable tags (hashes),
+  this gives flightctl the ability to check images being deployed on devices before the device could auto pick-up
+  a floating tag. (i.e. checks verifying the installation of the flightctl agent, and configuration)
+* When later in time a device has labels updated, the device controller will queue the device to be re-evaulated,
+  for example if a change of fleet owner happens because of the label changes.
+* An `annotation` is added to the device in the form of `fleet-controller/templateVersion` to track the latest device
+  template applied to the device.
+
+TemplateVersions are exposed via the API as read-only objects, they can be used to track changes to the fleet
+spec and the external references that were frozen on the devices:
+
+* `/fleet/{name}/templateversions` - This endpoint enables listing of `TemplateVersion` objects for the fleet.
+* `/fleet/{name}/templateversions/{version}` - This endpoint returns a specific `TemplateVersion` object for the fleet.
+
+#### Device management within a fleet
+
+##### Removing a device from a fleet
+
+A device could be stripped from the labels associating it to a specific fleet, in such
+case the `metadata.owner` field will be cleared by the fleet controller, and the device
+will need to be manually manage or eventually associated to a different fleet/owner.
+
+##### Changing a device to a different fleet
+
+A device could be moved to a different fleet, in such case the `metadata.owner` field will be updated
+by the fleet controller, and the device will be managed by the new fleet.
+
+This can be performed by modifying device labels to match a new fleet or modifying the fleet's device
+selector.
+
+##### Temporarily disabling fleet management (Fleet Device paused)
+
+A user can temporarily stop fleet management of a device by setting the `flightctl.io/fleet-controller=paused`
+in the device labels, the fleet manager will ignore devices with this label for any subsequent
+update or rollout.
+
+This feature is intended to allow temporarily debugging a device without de-associating
+from the fleet. This will mean the `metadata.owner` field is not cleared.
```

Comment:
This means the device will be guaranteed to not receive any templateVersion update from the service if paused?

Reply:
Right.